### PR TITLE
HPCC-23450 Security Manager to determine LDAP user account status

### PR DESCRIPTION
**Pull request by RussWhitehead**
In order to help admins know why a user cannot log into their LDAP account,
try to determine the nature of the error and return it back to ECLWatch/ECLIDE

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [x] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@wangkx Please review

File: esp/bindings/http/platform/httpservice.cpp [hunk 1]
```diff
@@ -2036,18 +2036,31 @@ void CEspHttpServer::logoutSession(EspAuthRequest& authReq, unsigned sessionID,
...
+        case AS_ACCOUNT_DISABLED :
+            ESPLOG(LogMin, "Account disabled for %s", authReq.ctx->queryUserId());
+            break;
+        case AS_ACCOUNT_EXPIRED :
+            ESPLOG(LogMin, "Account expired for %s", authReq.ctx->queryUserId());
+            break;
+        case AS_ACCOUNT_LOCKED :
+            ESPLOG(LogMin, "Account locked for %s", authReq.ctx->queryUserId());
+            break;
+        }
```
Comment (wangkx) [inline esp/bindings/http/platform/httpservice.cpp:2061]: Beside the ESPLOGs, we also need to inform ECLWatch the account status. For example, add:
addCookie("ESPAuthenticationAccountMSG", msg, 0, false);

**Review by wangkx — COMMENTED** [review]
@RussWhitehead one comment

File: esp/bindings/http/platform/httpservice.cpp (hunk 1 above)
Comment (RussWhitehead) [reply esp/bindings/http/platform/httpservice.cpp:2061]: @wangkx  Doesn't ECLWatch get notified via the "accountStatus" variable added to the myAccount response and set in ws_accountService.cpp below?

Comment (wangkx) [reply esp/bindings/http/platform/httpservice.cpp:2061]: Will ECLWatch call the ws_account.MyAccount whenever a user tries to log in? I believe not. @miguelvazq ?

Comment (miguelvazq) [reply esp/bindings/http/platform/httpservice.cpp:2061]: No it only calls once authenticated and ESP serves up the src directory.

**Comment by RussWhitehead** [comment]
@wangkx Please rereview

**Review by wangkx — APPROVED** [review]
@RussWhitehead looks fine

**Pull request merged**